featuretools/tests/entityset_tests/test_serialization.py

```diff
@@ -7,207 +7,167 @@
     _es = deserialize.description_to_entityset(description)
-    assert dask_es.metadata.__eq__(_es, deep=True)
+    assert es.__eq__(_es, deep=True)
 
 
-def test_invalid_formats(es, tmpdir):
+# TODO: Add test that checks serialization with a custom WW logical type
+def test_with_custom_ww_logical_type():
+    class CustomLogicalType(ww.logical_types.LogicalType):
+        pass
+
+    ww.type_system.add_type(CustomLogicalType)
+    columns = ['integer', 'natural_language', 'custom_logical_type']
+    dataframe = pd.DataFrame(columns=columns)
+    es = EntitySet()
+    ltype_dict = {
+        'integer': 'integer',
+        'natural_language': 'natural_language',
+        'custom_logical_type': CustomLogicalType,
+    }
+    es.add_dataframe(dataframe=dataframe, dataframe_name='custom_type', index='integer', logical_types=ltype_dict)
+    description = serialize.entityset_to_description(es)
+    _es = deserialize.description_to_entityset(description)
+    assert isinstance(_es['custom_type'].ww.logical_types['custom_logical_type'], CustomLogicalType)
+    assert es.__eq__(_es, deep=True)
+
+
+def test_serialize_invalid_formats(es, tmpdir):
     error_text = 'must be one of the following formats: {}'
     error_text = error_text.format(', '.join(serialize.FORMATS))
     with pytest.raises(ValueError, match=error_text):
-        serialize.write_entity_data(es.entities[0], path=str(tmpdir), format='')
-    with pytest.raises(ValueError, match=error_text):
-        entity = {'loading_info': {'location': 'data', 'type': ''}}
-        deserialize.read_entity_data(entity, path='.')
+        serialize.write_data_description(es, path=str(tmpdir), format='')
 
 
 def test_empty_dataframe(es):
-    for entity in es.entities:
-        description = serialize.entity_to_description(entity)
+    for df in es.dataframes:
+        description = typing_info_to_dict(df)
         dataframe = deserialize.empty_dataframe(description)
         assert dataframe.empty
+        assert all(dataframe.columns == df.columns)
 
 
+# TODO: Verify this works after WW Issue #950 is closed
 def test_to_csv(es, tmpdir):
     es.to_csv(str(tmpdir), encoding='utf-8', engine='python')
     new_es = deserialize.read_entityset(str(tmpdir))
     assert es.__eq__(new_es, deep=True)
-    df = to_pandas(es['log'].df, index='id')
-    new_df = to_pandas(new_es['log'].df, index='id')
+    df = to_pandas(es['log'], index='id')
+    new_df = to_pandas(new_es['log'], index='id')
     assert type(df['latlong'][0]) in (tuple, list)
     assert type(new_df['latlong'][0]) in (tuple, list)
 
 
-# Dask does not support to_pickle
+# Dask/Koalas don't support auto setting of interesting values with es.add_interesting_values()
+def test_to_csv_interesting_values(pd_es, tmpdir):
+    pd_es.add_interesting_values()
+    pd_es.to_csv(str(tmpdir))
+    new_es = deserialize.read_entityset(str(tmpdir))
+    assert pd_es.__eq__(new_es, deep=True)
+
+
+# TODO: Verify this works after WW Issue #950 is closed
```

Comment:
able to remove this? Or waiting until Featuretools is using Woodwork 0.4.1?

Reply:
I did test this works with the WW code on main, so I will go ahead and remove this comment. Will just need to make sure to update WW minimum version to 0.4.1 once it is released. I can make an issue so we don't forget about that.